packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/common.rs

```diff
@@ -225,6 +225,42 @@ pub trait ComponentAttributes: ComponentNode {
     }
 }
 
+/// Trait that creates state variables from attribute variants.
+/// For example, if implemented on the enum `Attrs`
+///
+/// ```rust
+/// enum Attrs {
+///   Prefill,
+///   Disabled,
+/// }
+/// ```
+/// one can call `Attrs::Prefill.state_var()` to get a state variable that will query the `"prefill"`
+/// attribute of a component.
+///
+/// An implementation might look like
+/// ```rust
+/// impl AttributeStateVar<String> for Attrs {
+///   fn state_var(&self) -> StateVar<String> {
+///     match self {
+///       Attrs::Prefill => StringStateVar::new_from_attribute("prefill", "").into_state_var(),
+///       _ => panic!("This attribute does not have a string state variable."),
+///     }
+///   }
+/// }
+/// impl AttributeStateVar<bool> for Attrs {
+///   fn state_var(&self) -> StateVar<bool> {
+///     match self {
+///       Attrs::Disabled => BooleanStateVar::new_from_attribute("disabled", false).into_state_var(),
+///       _ => panic!("This attribute does not have a boolean state variable."),
+///     }
+///   }
+/// }
+/// ```
+pub trait AttributeStateVar<T: Default + Clone> {
```

Comment:
I'd like to see the `default` more obviously specified in the interface. Whenever you implement an attribute state variable, you have to decide on a default in order to handle the case when no attribute was specified on the component.

Maybe the method should be
```
  fn state_var(&self, default: T) -> StateVar<T>;
```
Then, the example implementations would use `default` rather than `""` and `false`.

Reply:
And alternative is that you specify the default on the attribute `enum`, and then it doesn't need to be specified when you create the state variable definition. (I'm on the fence when is the best time to specify defaults.) It looks you are specifying defaults in our `TextInputAttribute`, so I'm not yet sure how they are working together.

Reply:
I'm not understanding your comment...Are you commenting on the docstring? The default is speficied on `TextInputAttribute` fields and nowhere else. A macro copies it to the correct location.

Reply:
That's right. I was thinking about implementing this manually. Now, I see you that you are implementing it via the macro and using the defaults from the macro attributes.

In this case, since you have the defaults defined in the attribute itself, I see why you don't want to have the default as an argument to the `state_var` method.